packages/router/src/RouterLink.ts

```diff
@@ -93,7 +95,40 @@ export function useLink(props: UseLinkOptions) {
   const router = inject(routerKey)!
   const currentRoute = inject(routeLocationKey)!
 
-  const route = computed(() => router.resolve(unref(props.to)))
+  const isValidTo = (to: unknown) => typeof to === 'string' || isObject(to)
```

Comment:
Let's inline this and `isObject` checks

Reply:
I can make this change, but I want to make sure we're on the same page before I do.

If I've understood correctly, you're proposing removing the `isValidTo` and `isObject` functions and just writing the code inline instead?

- The `isValidTo` function is used in two places, so I'd have to inline it twice.
- `isObject` is used in two places, but if I inline `isValidTo` too then I'd need to inline `isObject` in three places.

`isObject` could potentially be used in various other places. That seemed like a separate refactoring so I didn't do it in this PR, but we have various places with `typeof` checks for `object`.

I've also just noticed the `isRouteLocation` function, which seems to be doing that same thing as `isValidTo`:

- https://github.com/vuejs/router/blob/63770832d01a86a7464fde744e94058aae07de2e/packages/router/src/types/typeGuards.ts#L3

Should I maybe be using that instead?

Reply:
You are right, using `isRouteLocation()` is even better.